src/control/cmd/dmg/storage.go

```diff
@@ -267,3 +269,28 @@ func (u *storageUpdateCmd) Execute(args []string) error {
 //
 //	return nil
 //}
+
+// storageSetFaultyCmd is the struct representing the set-faulty storage subcommand
+type storageSetFaultyCmd struct {
+	logCmd
+	connectedCmd
+	Devuuid string `short:"u" long:"devuuid" description:"Device/Blobstore UUID to query" required:"1"`
+}
+
+// Set the SMD device state of the given device to "FAULTY"
+func storageSetFaulty(log logging.Logger, conns client.Connect, uuid string) {
+	if uuid == "" {
+		log.Infof("Device UUID needs to be specified\n")
```

Comment:
Shouldn't this be returned as an error? I believe that as it is currently, this message will be logged at INFO but the command will exit with 0 status. For an interactive user this may be fine, but it is devilishly hard to troubleshoot these things in a scripting environment when errors don't result in a non-zero status.

Reply:
makes sense, will return with error

Reply:
actually I specified this as a "required" command, so maybe this check is overkill?

Reply:
yes if it's required you shouldn't need to check